src/charm.py

```diff
@@ -198,6 +216,68 @@ def _get_hydra_endpoint_info(self) -> Optional[str]:
 
         return oauth2_provider_url
 
+    def _get_juju_config_identity_schema_config_(self) -> Optional[Tuple[str, Dict]]:
+        if identity_schemas := self.config.get("identity_schemas"):
+            default_schema = self.config.get("default_identity_schema")
+            if not default_schema:
+                logger.error(
+                    "`identity_schemas` configuration was set, but no `default_identity_schema` was found"
+                )
+                logger.warning("Ignoring `identity_schemas` configuration")
+                return None
+
+            try:
+                schemas = json.loads(identity_schemas)
+            except json.JSONDecodeError as e:
+                logger.error(f"identity_schemas is not a valid json: {e}")
+                logger.warning("Ignoring `identity_schemas` configuration")
+                return None
+
+            # We order the dict items to make sure the dict is ordered always has the same order
+            returned_schemas = {
```

Comment:
I'd avoid the `returned` in variable names. This also applies elsewhere in this PR.